query/src/api/rpc/exchange/exchange_manager.rs

```diff
@@ -95,7 +97,8 @@ impl DataExchangeManager {
 
     // Execute query in background
     pub fn execute_partial_query(&self, query_id: &str) -> Result<()> {
-        let mut queries_coordinator = self.queries_coordinator.lock();
+        let queries_coordinator_lock = self.queries_coordinator.lock();
+        let queries_coordinator = unsafe { &mut *queries_coordinator_lock.as_ptr() };
```

Comment:
This is almost equivalent to https://github.com/datafuselabs/databend/pull/6568/files#diff-b071b784b4bf25a377c65303683142fab21ad983b9ab57834a10fcc0ed3898c6

```
pub struct ReentrantMutexGuard<'a, T> {
    inner: ParkingReentrantMutexGuard<'a, UnsafeCell<T>>,
}

impl<'a, T> DerefMut for ReentrantMutexGuard<'a, T> {
    fn deref_mut(&mut self) -> &mut Self::Target {
        unsafe { &mut *self.inner.deref().get() }
    }
}
```

but I don't think we need to achieve it anymore.

Reply:
cc @zhang2014

Reply:
Great job.